src/transformers/modeling_bart.py

```diff
@@ -590,11 +593,12 @@ def forward(
             if use_cache:
                 next_decoder_cache.append(layer_past.copy())
 
-            if self.layer_norm and (idx == len(self.layers) - 1):  # if config.add_final_layer_norm (mBART)
```

Comment:
why check if you are at last step of loop when you can just wait for end of loop.